src/core_modules/capture-core/components/WidgetStagesAndEvents/Stages/Stage/StageDetail/hooks/helpers.js

```diff
@@ -25,10 +25,14 @@ const getEventStatus = (event: ApiEnrollmentEvent) => {
     }
 
     if (event.status === statusTypes.SCHEDULE) {
-        if (!event.scheduledAt || !daysUntilDueDate) {
+        if (!event.scheduledAt) {
             return { status: statusTypes.SCHEDULE, options: undefined };
         }
 
+        if (daysUntilDueDate === 0) {
+            return { status: statusTypes.SCHEDULE, options: 'today' };
```

Comment:
Is this the string that get's displayed in the badge? If so, do you think it makes sense to add translations?

Reply:
@eirikhaugstulen No, it isn't the string that ended up being displayed, it gets passed to this [function](https://github.com/dhis2/capture-app/blob/6970f4415ee059f5401802619cab751cf2810d5a/src/core_modules/capture-core/events/statusTypes.js#L16) here and get translated.

Reply:
The string is passed as a dynamic value into the [translation](https://github.com/dhis2/capture-app/blob/6970f4415ee059f5401802619cab751cf2810d5a/src/core_modules/capture-core/events/statusTypes.js#L17). As far as I understand from the [i18next interpolation docs](https://www.i18next.com/translation-function/interpolation), a dynamic value is not translated. When I change the language to French, this is what is currently displayed:
1. <img width="226" alt="Screenshot 2024-10-23 at 17 46 06" src="https://github.com/user-attachments/assets/e667f253-a565-4210-bf6d-bbae3fd93411">

By adding translations `option: i18n.t('Today')` as Eirik suggested, the text is translated correctly:

2. <img width="237" alt="Screenshot 2024-10-23 at 17 19 49" src="https://github.com/user-attachments/assets/69cd2f39-ddcf-4ce2-b7db-4a6ce2d5b097">

Could you please recheck @alaa-yahia? Let me know if you need any help or have doubts. Thank you!


PS: For these examples, I deliberately chose 'Today' in uppercase because the string already exists translated in the language files, but you can also use lowercase and Transfix will add the translations in a few weeks.

Reply:
Ah whoops! I noticed dueDateFromNow was being translated, so I figured the same would happen with today's date. Didn't realize moment.js has a localization.